pkg/server/options/controllers.go

```diff
@@ -55,6 +78,33 @@ func (c *Controllers) AddFlags(fs *pflag.FlagSet) {
 	apiimporter.BindOptions(&c.ApiImporter, fs)
 	apiresource.BindOptions(&c.ApiResource, fs)
 	syncer.BindOptions(&c.Syncer, fs)
+
+	c.SAController.AddFlags(fs)
+}
+
+func (c *Controllers) Complete(rootDir string) error {
+	if c.SAController.ServiceAccountKeyFile == "" {
+		// use sa.key and auto-generate if not existing
+		c.SAController.ServiceAccountKeyFile = filepath.Join(rootDir, "sa.key")
+		if _, err := os.Stat(c.SAController.ServiceAccountKeyFile); os.IsNotExist(err) {
+			klog.Infof("Generating service account key file %s", c.SAController.ServiceAccountKeyFile)
+			// TODO(ncdc): should alrogithm & key size be configurable?
+			key, err := rsa.GenerateKey(cryptorand.Reader, 2048)
```

Comment:
(No action required) Maybe update this to 4096 and remove the comment above? I think rsa+4096 should be good enough until we learn otherwise.

Reply:
done